### PR TITLE
Add ability to open PR as draft to Submit-PullRequest.ps1

### PR DESCRIPTION
**Pull request by danieljurek**
Give the ability to open PRs as draft.

### CODE REVIEW

File: eng/common/scripts/Submit-PullRequest.ps1 [hunk 1]
```diff
@@ -79,9 +94,16 @@ if ($resp.Count -gt 0) {
...
-    -AuthToken $AuthToken
+    $resp = New-GitHubPullRequest `
+      -RepoOwner $RepoOwner `
+      -RepoName $RepoName `
+      -Title $PRTitle `
+      -Head "${PROwner}:${PRBranch}" `
+      -Base $BaseBranch `
+      -Body $PRBody `
+      -Maintainer_Can_Modify $true `
+      -Draft:$Draft `
```
Comment (danieljurek) [inline eng/common/scripts/Submit-PullRequest.ps1:105]: Formatting changed, this is the added line.

File: eng/common/pipelines/templates/steps/create-pull-request.yml [hunk 2]
```diff
@@ -87,3 +88,4 @@ steps:
       -TeamReviewers "${{ parameters.GHTeamReviewersVariable }}"
       -Assignees "${{ parameters.GHAssignessVariable }}"
       -CloseAfterOpenForTesting $${{ coalesce(parameters.CloseAfterOpenForTesting, 'false') }}
+      -Draft $${{ coalesce(parameters.DraftPr, 'false') }}
```
Comment (danieljurek) [inline eng/common/pipelines/templates/steps/create-pull-request.yml:91]: Following the lead here with `coalesce` though I'm not sure it's necessary.

Comment (weshaggard) [reply eng/common/pipelines/templates/steps/create-pull-request.yml:91]: LOL I was going to ask the exact question I don't think it is necessary in either case.

File: eng/common/scripts/Submit-PullRequest.ps1 [hunk 3]
```diff
@@ -56,7 +69,9 @@ param(
 
   [string]$Assignees,
 
-  [boolean]$CloseAfterOpenForTesting=$false
+  [boolean]$CloseAfterOpenForTesting=$false,
+
+  [boolean]$Draft=$false
```
Comment (weshaggard) [inline eng/common/scripts/Submit-PullRequest.ps1:74]: Can we call this $OpenAsDraft?

File: eng/common/pipelines/templates/steps/create-pull-request.yml [hunk 4]
```diff
@@ -20,6 +20,7 @@ parameters:
   PRLabels: ''
   SkipCheckingForChanges: false
   CloseAfterOpenForTesting: false
+  DraftPr: false
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/create-pull-request.yml:23]: Can we call this OpenAsDraft?

**Review by weshaggard — APPROVED** [review]
Couple suggestions but otherwise looks good. Thanks for adding this support.

**Comment by danieljurek** [comment]
Removed auto-merge tag. Following the actual process :)

File: eng/common/scripts/Submit-PullRequest.ps1 [hunk 5]
```diff
@@ -21,6 +21,19 @@ The title of the pull request.
 The body message for the pull request. 
 .PARAMETER PRLabels
 The labels added to the PRs. Multple labels seperated by comma, e.g "bug, service"
+.PARAMETER UserReviewers
+User reviewers to request after opening the PR. Users should be a comma-
+separated list with no preceeding `@` symbol (e.g. "user1,usertwo,user3")
+.PARAMETER TeamReviewers
+TODO
```
Comment (weshaggard) [inline eng/common/scripts/Submit-PullRequest.ps1:28]: List of github teams to add ass reviewers.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**